tests/e2e/data_masking/test_data_masking.py

```diff
@@ -0,0 +1,129 @@
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    # AWS Encryption SDK encrypt method only takes in bytes or strings
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting and then decrypting the encrypted data
+    encrypted_data = data_masker.encrypt(value)
+    decrypted_data = data_masker.decrypt(encrypted_data)
+
+    # THEN the result is the original input data
+    assert decrypted_data == value
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_context(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting and then decrypting the encrypted data with an encryption_context
+    encrypted_data = data_masker.encrypt(value, encryption_context={"this": "is_secure"})
+    decrypted_data = data_masker.decrypt(encrypted_data, encryption_context={"this": "is_secure"})
+
+    # THEN the result is the original input data
+    assert decrypted_data == value
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_diff_context_fail(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting with a encryption_context
+    encrypted_data = data_masker.encrypt(value, encryption_context={"this": "is_secure"})
+
+    # THEN decrypting with a different encryption_context should raise a ValueError
+    with pytest.raises(ValueError):
+        data_masker.decrypt(encrypted_data, encryption_context={"not": "same_context"})
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_no_context_fail(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting with no encryption_context
+    encrypted_data = data_masker.encrypt(value)
+
+    # THEN decrypting with an encryption_context should raise a ValueError
+    with pytest.raises(ValueError):
+        data_masker.decrypt(encrypted_data, encryption_context={"this": "is_secure"})
+
+
+# TODO: metaclass?
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_key_fail(data_masker, kms_key2_arn):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider with a certain key
+
+    # WHEN encrypting and then decrypting the encrypted data
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+    encrypted_data = data_masker.encrypt(value)
+
+    # THEN when decrypting with a different key it should fail
+    data_masker_key2 = DataMasking(provider=AwsEncryptionSdkProvider(keys=[kms_key2_arn]))
+
+    with pytest.raises(DecryptKeyError):
+        data_masker_key2.decrypt(encrypted_data)
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encrypted_in_logs(data_masker, basic_handler_fn, basic_handler_fn_arn):
```

Comment:
Suggestion to help us catch other types of issues here

* Use DataMasking within the Lambda Handler to encrypt whatever event we send
* Lambda to return the event encrypted
* Within the test, try to decrypt and compare with the original value

This will also help us debug potential issues that would only occur within the Lambda runtime (e.g., lack of CPU cycles to handle it, insufficient memory, Lambda's AWS SDK provided continues to work, etc.).

It will simplify your assertions too since traversing logs require a loop here, whereas the value is ensuring DataMasking works within any Lambda function with default settings.

Reply:
Implemented this test and then ran into the issue described in Leandro's comment below haha. I pushed a commit with the test anyway with a note that it doesn't work yet due to the dependency issue.